core/services/pipeline/task.jsonparse.go

```diff
@@ -110,12 +109,9 @@ func (t *JSONParseTask) Run(_ context.Context, l logger.Logger, vars Vars, input
 		}
 	}
 
-	switch val := decoded.(type) {
-	case json.Number:
-		decoded, err = getJsonNumberValue(val)
-		if err != nil {
-			return Result{Error: multierr.Combine(ErrBadInput, err)}, runInfo
-		}
+	decoded, err = reinterpetJsonNumbers(decoded)
```

Comment:
Huh, so this was already defined but we weren't using it?

Reply:
`reinterpetJsonNumbers` was a new function I introduced recently, just did not update this place.

Reply:
I cannot see it in this PR

Reply:
https://github.com/smartcontractkit/chainlink/pull/7434/files#diff-58294c479ec0dfd4ae41124d476e13a9d93777b46cd32c5f46d47eaaba442d0e